### PR TITLE
fix(filter-field): Fixes an issue with inconsistent clearAll functionality

### LINKED ISSUE #435: Filter field: Clear all button requires 2 clicks
# Bug Report

## Expected Behavior

"Clear all" should clear all (editable) fields on a single click

## Current Behavior

I have to click "Clear all" twice to remove the fields
(In some cases the button never has a (visual) effect no matter how often I click it until I scroll, but I can't reproduce this consistantly, so hopefully it's the same problem)

## Steps to Reproduce

https://barista.dynatrace.com/components/filter-field/#readonly-non-deletable--non-editable-tags

1. Add an editable field to the filter-field
2. Click "Clear all"

### PR DESCRIPTION
**Pull request by tomheller**
The clearAll button rendering depended on the filterfield not being
focussed. When the clearAll button was rendered and was clicked, the
cdkFocusMonitor kicked in, set the focus and removed the clearAll button
again. As a result the clearAll button lost its focus, the document
focus was set back to the body and the clearAll was rendered again. The
click event on the clearAll was never fired.
We no longer remove the clearAll from the DOM, but hide it via css
classes. This makes it not lose focus, and results in a more consistent
behavior.

Fixes #435

### <strong>Pull Request</strong>

<hr>
Hi, thank you for contributing to Barista with this pull request (PR).

To ensure a fast process and merging of your PR please make sure it fulfills the
coding standards and contribution guidelines.

- A feature proposal has been provided, discussed and approved first.
- There is a meaningful description of the issue in GitHub (Screenshots are
  often helpful).
- If the PR introduces breaking-changes or deprecations it matches the following
  guidelines.
  - The commit message follows our commit guidelines.
  - Tests for the changes have been added (for bug fixes / features).
  - Docs have been added / updated (for bug fixes / features).

Please choose the type appropriate for the changes below: <br>

#### Checklist

- [ ] I have read the CONTRIBUTING doc and I follow the PR guidelines
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: apps/components-e2e/src/components/filter-field/filter-field.e2e.ts [hunk 1]
```diff
@@ -76,3 +77,30 @@ test('should remove all filters when clicking the clear-all button', async (test
...
+  // // Click somewhere outside so the clear-all button appears
+  await testController.click(Selector('.outside'));
+  // Wait for the filterfield to catch up.
+  await testController.wait(500);
+  await testController.expect(await clearAll.exists).ok();
+
+  // Click the clear all-button, the created filter should be removed
+  await testController.click(clearAll);
+  // Wait for the filterfield to catch up.
+  await testController.wait(500);
```
Comment (lukasholzer) [inline apps/components-e2e/src/components/filter-field/filter-field.e2e.ts]: is this related to animations?

Maybe we should consider using the `NoopAnimationsModule` instead of the `BrowseranimationsModule`?
@ffriedl89 Maybe this would reduce flakiness?

File: apps/components-e2e/src/components/filter-field/filter-field.ts [hunk 2]
```diff
@@ -57,10 +58,75 @@ const TEST_DATA: DtFilterFieldDefaultDataSourceType = {
...
+
+  setupSecondTestScenario(): void {
+    this._dataSource = new DtFilterFieldDefaultDataSource(DATA[1]);
+    const linzFilter = [
+      DATA[1].autocomplete[0],
+      DATA[1].autocomplete[0].autocomplete![0],
+    ];
+
+    this._filterfield.filters = [linzFilter];
+    this._filterfield.currentTags.subscribe(() => {
```
Comment (lukasholzer) [inline apps/components-e2e/src/components/filter-field/filter-field.ts]: please unsubscribe in a `ngOnDestroy`

File: components/filter-field/src/filter-field.scss [hunk 3]
```diff
@@ -110,6 +110,11 @@ button.dt-filter-field-clear-all-button {
   margin-left: 8px;
 }
 
+button.dt-filter-field-clear-all-button-hidden {
```
Comment (lukasholzer) [inline components/filter-field/src/filter-field.scss]: why we have to hide it? Can we just `ngIf` it, please explain it to me :)

Comment (tomheller) [reply components/filter-field/src/filter-field.scss]: As seen in the description of the PR, if we *ngIf the button, it will interfere with how the filterfield uses the focusmonitor to show the button or not. The button would be removed on mouseDown, would immediately loose focus (because it is being removed), and be shown again, without the click handler ever firing.

> The clearAll button rendering depended on the filterfield not being
focussed. When the clearAll button was rendered and was clicked, the
cdkFocusMonitor kicked in, set the focus and removed the clearAll button
again. As a result the clearAll button lost its focus, the document
focus was set back to the body and the clearAll was rendered again. The
click event on the clearAll was never fired.
We no longer remove the clearAll from the DOM, but hide it via css
classes. This makes it not lose focus, and results in a more consistent
behavior.

File: apps/components-e2e/src/components/filter-field/filter-field.e2e.ts (hunk 1 above)
Comment (tomheller) [reply apps/components-e2e/src/components/filter-field/filter-field.e2e.ts]: It's not related to the animations module, but the way the filter field updates, it waits for the `zone` to get stable, before it will update it's state.

**Comment by tomheller** [comment]
I have resolved all conflicts. PR should be ready now.
